flocker/node/agents/blockdevice.py

```diff
@@ -983,6 +1007,33 @@ def create_volume_with_profile(name, size, profile_name):
         """
 
 
+@implementer(IProfiledBlockDeviceAPI)
+class ProfiledBlockDeviceAPIAdapter(PClass):
+    """
+    Adapter class to create ``IProfiledBlockDeviceAPI`` providers for
+    ``IBlockDeviceAPI`` implementations that do not implement
+    ``IProfiledBlockDeviceAPI``
+
+    :ivar _blockdevice_api: The ``IBlockDeviceAPI`` provider to back the
+        volume creation.
+    """
+    _blockdevice_api = field(
+        mandatory=True,
+        invariant=lambda i: (IBlockDeviceAPI.providedBy(i),
+                             '_blockdevice_api must provide IBlockDeviceAPI'))
+
+    def create_volume_with_profile(self, dataset_id, size, profile_name):
+        """
+        Reverts to constructing a volume with no profile. To be used with
+        backends that do not implement ``IProfiledBlockDeviceAPI``, but do
+        implement ``IBlockDeviceAPI``.
+        """
+        CREATE_VOLUME_PROFILE_DROPPED(dataset_id=dataset_id,
```

Comment:
In new code, use the implicit global Eliot logger, unless some interaction with existing code makes it trickier to do so.

In this case I think you can just `.write()` and update the tests to remove the monkey-patching.